### PR TITLE
Use local store for querying/listing events

### PR DESCRIPTION
**Pull request by m1l4n54v1c**
Use local event store (if possible) for list and query events methods (depending on the request).

Relates to https://github.com/AxonIQ/axon-server-api/pull/20

### CODE REVIEW

File: axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java [hunk 1]
```diff
@@ -312,35 +309,79 @@ public void getFirstToken(GetFirstTokenRequest request, StreamObserver<TrackingT
...
         );
     }
 
-
     public StreamObserver<QueryEventsRequest> queryEvents(StreamObserver<QueryEventsResponse> responseObserver0) {
-        ForwardingStreamObserver<QueryEventsResponse> responseObserver = new ForwardingStreamObserver<>(logger, "queryEvents", responseObserver0);
-        return checkConnection(contextProvider.getContext(), responseObserver).map(client -> client.queryEvents(contextProvider.getContext(), responseObserver)).orElse(null);
+        String context = contextProvider.getContext();
+        EventStore leaderEventStore = eventStoreLocator.getEventStore(context, false);
+        EventStore localEventStore = eventStoreLocator.getEventStore(context, true);
```
Comment (saratry) [inline axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java]: The name of this variable could be misleading, as the method may not return the local store but the remote one, according to the documentation.

File: axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java [hunk 2]
```diff
@@ -295,13 +291,14 @@ public void onCompleted() {
...
-        checkConnection(contextProvider.getContext(), responseObserver).ifPresent(client ->
+        checkConnection(contextProvider.getContext(), responseObserver, false).ifPresent(client ->
             client.getFirstToken(contextProvider.getContext(), request, responseObserver)
         );
     }
 
-    private Optional<EventStore> checkConnection(String context, StreamObserver<?> responseObserver) {
-        EventStore eventStore = eventStoreLocator.getEventStore(context);
+    private Optional<EventStore> checkConnection(String context, StreamObserver<?> responseObserver,
+                                                 boolean useLocalStore) {
```
Comment (saratry) [inline axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java]: It is always `false`, right? Maybe we can evaluate if the parameter is really needed.

File: axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java (hunk 1 above)
Comment (m1l4n54v1c) [reply axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java]: You are right! But I'm not sure about the name :) `maybeLocalEventStore`? 😄

File: axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java (hunk 2 above)
Comment (m1l4n54v1c) [reply axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java]: This is a leftover. The first implementation involved much more gRPC methods. Will fix it!

File: axonserver/src/main/java/io/axoniq/axonserver/topology/DefaultEventStoreLocator.java [hunk 3]
```diff
@@ -37,10 +39,10 @@ public boolean isLeader(String nodeName, String contextName, boolean waitForLead
     }
 
     @Override
-    public EventStore getEventStore(String context) {
-        if( Topology.DEFAULT_CONTEXT.equals(context))
+    public EventStore getEventStore(String context, boolean useLocal) {
```
Comment (MGathier) [inline axonserver/src/main/java/io/axoniq/axonserver/topology/DefaultEventStoreLocator.java]: allowFollower

**Pull request merged**